### ISSUE TITLE
"Browsing by Issue Date" year suppression is confusing as some years are not obviously accessible

### ISSUE THREAD

**Issue by dsteelma-umd** [DSpace/dspace-angular#3374]
## Is your feature request related to a problem? Please describe.

The "(Choose year)" drop-down on the "Browsing by Issue Date" intentionally suppresses years as it goes further back in time, as per the following comment in [browse-by-date.component.ts](https://github.com/DSpace/dspace-angular/blob/3e6adbba9359ad5740581716fd45402b2a620a45/src/app/browse-by/browse-by-date/browse-by-date.component.ts#L129-L131):

```
   * Update the StartsWith options
   * In this implementation, it creates a list of years starting from the most recent item or the current year, going
   * all the way back to the earliest date found on an item within this scope. The further back in time, the bigger
   * the change in years become to avoid extremely long lists with a one-year difference.
```

For example, in the DSpace demo site, https://demo.dspace.org/browse/dateissued, the drop-down shows the following years:

* 2024
* 2023
* ...
* 2011
* 2010
* 2005
* 2000
* ...

Selecting "2005" returns no items. There is an item for the year "2006", specifically https://demo.dspace.org/entities/datafile/fa0931d6-49db-4027-829b-c1ebe84f2f3f, but the only way to display the items for the year 2006 is to enter `2006` into the "Filter results by date" textbox. If does not appear possible to access items published in 2006 via the "(Choose year)" drop-down.

As it stands, the functionality is extremely confusing on two levels:

* For users, it is not clear how to search for items in the suppressed years
* For curators, it looks like items that have a "dc.date.issued" field in the suppressed years are not being properly indexed, because the year is not displayed in the drop-down.

## To Demonstrate

To demonstrate on the DSpace demo site:

1. In a web browser, go to

    https://demo.dspace.org/browse/dateissued

    The "Browsing by Issue Date" page will be displayed..

2. On the "Browsing by Issue Date" page, left-click the "(Choose year)" drop-down and note that there is a "2010" entry immediately followed by "2005", i.e., there is no "2006" entry. Left-click the "2005" entry. The page will update and indicate that there are no items to show.

3. Type `2006` into the "Filter results by date" textbox, and left-click the "Browse" button. A single result "Animation Movies Trailer Computation" will be displayed.

## Describe the solution you'd like

The general expectation would be that selecting "2005" would be either the equivalent of searching through 2005-2009 (i.e., all the years that are suppressed), or it would be possible to "drill down" by having selected "2005" that the intervening years of 2006, 2007, 2008, and 2009 would be displayed so that they could be selected.

## Describe alternatives or workarounds you've considered

Entering the desired year directly in the "Filter results by date" textbox allows items for that year to be found, but it is not obvious to users that they need to do that for the suppressed years.

## Additional information

None.

**Comment by tdonohue** [comment]
I agree this sounds confusing for users.  I wasn't aware of this behavior, but agree it should be fixed.  Either we should show date ranges (for suppressed years) or we should find a way to show all years.

Moving to maintenance board, as this seems like a bug.  Needs a volunteer to investigate.
